subworkflows/local/sc_raw_input.nf

```diff
@@ -26,68 +26,64 @@ workflow SC_RAW_INPUT {
-        }
-        if (params.cprimers)  {
-            error "The transcript-specific primer, 5'-RACE, UMI library generation method does not require C-region primers, please provide a reference file instead or select another library method option."
-        }
-        if (params.umi_length > 0)  {
-            error "The transcript-specific primer, 5'-RACE, UMI library generation method does not require to set the UMI length, please provide a reference file instead or select another library method option."
-        }
-        if (params.reference_10x)  {
-            // necessary to allow tar.gz files as input so that tests can run
-            if (params.reference_10x.endsWith(".tar.gz")){
-                UNZIP_CELLRANGERDB(
-                    params.reference_10x
-                )
-                UNZIP_CELLRANGERDB.out.unzipped.set { ch_sc_reference }
-            } else {
-                ch_sc_reference = Channel.fromPath(params.reference_10x, checkIfExists: true)
-            }
+    if (params.vprimers) {
+        error "The single-cell 10X genomics library generation method does not require V-region primers, please provide a reference file instead or select another library method option."
+    } else if (params.race_linker) {
+        error "The single-cell 10X genomics library generation method does not require the --race_linker parameter, please provide a reference file instead or select another library method option."
+    }
+    if (params.cprimers)  {
+        error "The single-cell 10X genomics library generation method does not require C-region primers, please provide a reference file instead or select another library method option."
+    }
+    if (params.umi_length > 0)  {
+        error "The single-cell 10X genomics library generation method does not require to set the UMI length, please provide a reference file instead or select another library method option."
+    }
+    if (params.reference_10x)  {
+        // necessary to allow tar.gz files as input so that tests can run
+        if (params.reference_10x.endsWith(".tar.gz")){
+            UNZIP_CELLRANGERDB(
+                params.reference_10x
+            )
+            UNZIP_CELLRANGERDB.out.unzipped.set { ch_sc_reference }
         } else {
-            error "The transcript-specific primer, 5'-RACE, UMI library generation method requires you to provide a reference file."
+            ch_sc_reference = Channel.fromPath(params.reference_10x, checkIfExists: true)
         }
+    } else {
+        error "The single-cell 10X genomics library generation method requires you to provide a reference file."
+    }
 
-        // run cellranger vdj
-        CELLRANGER_VDJ (
+    // run cellranger vdj
+    CELLRANGER_VDJ (
         ch_reads,
-        ch_sc_reference
-        )
-        ch_versions = ch_versions.mix(CELLRANGER_VDJ.out.versions)
+        ch_sc_reference.collect()
```

Comment:
@mapo9 this one is important, if combining a channel with multiple input files, with one that only contains 1 file, e.g. the reference here, one needs to use the `.collect()` method for the reference, as otherwise the CELLRANGER_VDJ process will only be executed once for one sample at random.

Reply:
thanks! fixing this behaviour would've taken me hours probably